new git pull 12/03/2021 @ 12:29 eST

I am not running a RPiHQ camera but during the install I see the following I have a ZWO camera but wanted to share what I see.


capture_RPiHQ.cpp: In function ‘int main(int, char**)’:
capture_RPiHQ.cpp:829:14: warning: variable ‘ImgExtraText’ set but not used [-Wu                                                                                                                                                             nused-but-set-variable]
  char const *ImgExtraText   = "";
              ^~~~~~~~~~~~
capture_RPiHQ.cpp:830:6: warning: variable ‘extraFileAge’ set but not used [-Wun                                                                                                                                                             used-but-set-variable]
  int extraFileAge           = 0;   // 0 disables it

--- a/src/capture.cpp
+++ b/src/capture.cpp
@@ -626,7 +626,7 @@ ASI_ERROR_CODE takeOneExposure(
 	if (currentAutoExposure == ASI_TRUE && exposure_time_us > current_max_autoexposure_us)
 	{
 		// If we call length_in_units() twice in same command line they both return the last value.
-		char x[20];
+		char x[100];
 		snprintf(x, sizeof(x), "%s", length_in_units(exposure_time_us, true));
 		Log(0, "*** WARNING: exposure_time_us requested [%s] > current_max_autoexposure_us [%s]\n", x, length_in_units(current_max_autoexposure_us, true));
 		exposure_time_us = current_max_autoexposure_us;

--- a/src/capture_RPiHQ.cpp
+++ b/src/capture_RPiHQ.cpp
@@ -828,6 +828,8 @@ const char *locale         = DEFAULT_LOCALE;
 	char const *ImgText   = "";
 	char const *ImgExtraText   = "";
 	int extraFileAge           = 0;   // 0 disables it
+// The "extra text" file hasn't been implemented in RPiHQ.  The next line keeps the compiler quiet so users don't think there's a problem.
+if (extraFileAge == 99999 && ImgExtraText[0] == '\0') ImgExtraText = "xxxxxx   keep compiler quiet";
 #define DEFAULT_FONTSIZE    32
 	double fontsize       = DEFAULT_FONTSIZE;
 #define SMALLFONTSIZE_MULTIPLIER 0.08
@@ -1645,7 +1647,7 @@ const char *locale         = DEFAULT_LOCALE;
 				{
 					long last_exposure_us = currentExposure_us;
 
-					float actualGain = currentGain; // to be compatible with ZWO - ZWO gain=0.1 dB , RPiHQ gain=factor
+					float actualGain = currentGain;	// to be compatible with ZWO - ZWO gain=0.1 dB , RPiHQ gain=factor
 					if (myModeMeanSetting.mode_mean)
 						actualGain =  myRaspistillSetting.analoggain;
 					int iYOffset = 0;

